services/storage/src/simcore_service_storage/access_layer.py

```diff
@@ -0,0 +1,266 @@
+    """
+    user_group_ids: List[int] = await _get_user_groups_ids(conn, user_id)
+
+    stmt = text(
+        f"""\
+        SELECT prj_owner, access_rights
+        FROM projects
+        WHERE (
+            ( uuid = '{project_id}' ) AND (
+                prj_owner = {user_id}
+                OR jsonb_exists_any( access_rights, (
+                    SELECT ARRAY( SELECT gid::TEXT FROM user_to_groups WHERE uid = {user_id} )
+                    )
+                )
+            )
+        )
+        """
+    )
+
+    result: ResultProxy = await conn.execute(stmt)
+    row: Optional[RowProxy] = await result.first()
+
+    if not row:
+        # Either project does not exists OR user_id has NO access
+        return AccessRights.none()
+
+    assert isinstance(row.prj_owner, int)
+    assert isinstance(row.access_rights, dict)
+
+    if row.prj_owner == user_id:
+        return AccessRights.all()
+
+    # determine user's access rights by aggregating AR of all groups
+    prj_access = _aggregate_access_rights(row.access_rights, user_group_ids)
+    return prj_access
+
+
+async def get_file_access_rights(
+    conn: SAConnection, user_id: int, file_uuid: str
+) -> AccessRights:
+    """
+    Returns access-rights of user (user_id) over data file resource (file_uuid)
+    """
+
+    #
+    # 1. file registered in file_meta_data table
+    #
+    stmt = sa.select([file_meta_data.c.project_id, file_meta_data.c.user_id]).where(
+        file_meta_data.c.file_uuid == file_uuid
+    )
+    result: ResultProxy = await conn.execute(stmt)
+    row: Optional[RowProxy] = await result.first()
+
+    if row:
+        if int(row.user_id) == user_id:
+            # is owner
+            return AccessRights.all()
+
+        if not row.project_id:
+            # not owner and not shared via project
+            return AccessRights.none()
+
+        # has associated project
+        access_rights = await get_project_access_rights(
+            conn, user_id, project_id=row.project_id
+        )
+        if not access_rights:
+            logger.warning(
+                "File %s references a project %s that does not exists in db."
+                "TIP: Audit sync between files_meta_data and projects tables",
+                file_uuid,
+                row.project_id,
+            )
+            return AccessRights.none()
+
+    else:
+        #
+        # 2. file is NOT registered in meta-data table e.g. it is about to be uploaded or it was deleted
+        #    We rely on the assumption that file_uuid is formatted either as
+        #
+        #       - project's data: {project_id}/{node_id}/{filename}
+        #       - API data:       api/{file_id}/{filename}
+        #
+        try:
+            parent, _, _ = file_uuid.split("/")
+
+            if parent == "api":
+                # ownership still not defined, so we assume it is user_id
+                return AccessRights.all()
+
+            _ = UUID(parent)  # tests parent as UUID
```

Comment:
I would go with this one.
```suggestion
            UUID(parent)  # raises ValueError in case parent is not a valid uuid
```